src/engine.rs

```diff
@@ -32,6 +32,18 @@ struct PromptWidget {
     origin: (u16, u16),
 }
 
+enum ReedlineEvent {
```

Comment:
this is the main part of the design that I was thinking every Event generates this enum (definitely needs a better name). This will eventually help in decluttering the main struct yet be functionally equivalent. 

Clock poll generates an events
emacs mode generates events
vi mode generates events

then all these are handled in the same function. I want to make distinction between this and EditCommand. This is line-editor events (i.e. everything) where as I was going to reduce `EditCommand`s to core-editor specific operations only.